src/build/helpers.ts

```diff
@@ -62,6 +62,8 @@ export const baseTypeConversionMap = new Map<string, string>([
   ["EventHandler", "EventHandler"],
 ]);
 
+export const genericEventSupertypes = new Set(["UIEvent", "MouseEvent"]);
```

Comment:
I expected this to cover every events, why should it be limited?

Reply:
It will allow to support dispatched events in future (just idea).

```javascript
const event /* : MouseEvent<never> */ = new MouseEvent("click")

button.addEventListener("ckick", (event /* : MouseEvent<HTMLButtonElement> */) => {})
```

Also important - `CustomEvent`, `MessageEvent` and `ProgressEvent` already use `T` type parameter. Could we use other type parameter name for `currentTarget`?

Reply:
@saschanaz
> I expected this to cover every event, why should it be limited?

I just thought it might be useful to be able to limit the implementation to DOM events in case performance became an issue, but I guess we could apply this to every `Event` subtype, if you think it is better.

@turansky 
>  CustomEvent, MessageEvent and ProgressEvent already use T type parameter. Could we use other type parameter name for currentTarget?

I was being conservative about the change, so the current implementation only takes interfaces that have no type parameter defined (I made it so there would not be unexpected conflicts) so, for those types we could override on the json files. But yes, we could instead always use another name if that seems better. Does something like `Target` sound good? I'm not sure how opinionated is the project regarding names.

Reply:
Updated request - use `T` for `currentTarget` type parameter in all events.

For `ProgressEvent` it will be common type parameter for both targets (now it's used for `target` only).

For `MessgegeEvent.data` and `CustomEvent.details` we can use `D` as type parameter name.

Reply:
> I just thought it might be useful to be able to limit the implementation to DOM events in case performance became an issue, but I guess we could apply this to every `Event` subtype, if you think it is better.

We should first check the difference and then decide based on that.

Reply:
@turansky @saschanaz Ok, I'll see to handle all events instead of whitelisting and will rename the mentioned type parameters to `D`.

In the meantime, I updated the PR with a possible approach to type the `onevent` handlers for the currently handled interfaces.

Reply:
@nscarcella FYI - I implemented similar changes in [Kotlin Wrappers](https://github.com/JetBrains/kotlin-wrappers/pull/2217).
I used followings script:
1. Remove `T` type parameter from `ProgressEvent`
  * To replace it later with common `C`
2. `D` type parameter for `CustomEvent` and `MessageEvent`
  * `T` reserved as future type parameter for `target` (will work fine in handlers like `FileReader.onload`)
3. `C` - type parameter for `currentTarget` for all events